File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.ts
function countPartitions(nums: number[]): number {
    let sum = nums.reduce((a, b) => a + b, 0);
    let currSum = 0;
    let partitions = 0;
    nums.forEach((val, idx) => {
        if (idx != 0) {
            currSum += val;
            partitions += Number((sum - 2 * currSum) % 2 == 0);
        }
    });
    return partitions;
};

File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.go
func countPartitions(nums []int) int {
    sum := 0;
    for _, val := range(nums) {
        sum += val
    }

    currSum := 0
    partitions := 0
    for idx, val := range(nums) {
        if idx == 0 {
            continue;
        }
        
        currSum += val
        if (sum - 2 * currSum) % 2 == 0 {
            partitions++;
        }
    }

    return partitions;
}